fetch_index.py

```diff
@@ -0,0 +1,354 @@
+                    return []
+            responses.append(response_data)
+        return responses
+
+    def get_lists_paths(self, sites, response_data=None):
+        logger.info('Extracting list name')
+        lists = {}
+        if not response_data:
+            logger.info(
+                'List response is not present. Fetching the list for sites')
+            response_data = self.index_lists(sites, index=False) or []
+
+        lists = {}
+        for response in response_data:
+            for result in response:
+                lists[result.get('title')] = result.get('parentweburl')
+
+        return lists
+
+    def index_items(self, lists, index):
+        responses = []
+        logger.info('Fetching all the items for the lists')
+        for list_content, value in lists.items():
+            rel_url = urljoin(
+                self.sharepoint_host, "{}/_api/web/lists/getbytitle(\'{}\')/items".format(list_content, value))
+
+            logger.info(
+                'Fetching the items for list: {} from url: {}'.format(list, rel_url))
+
+            # Condition to check if the list does not have a file/folder
+            if self.sharepoint_client.get(rel_url, "?$select=Folder Eq null") or self.sharepoint_client.get(rel_url, "?$select=File Eq null"):
+                logger.info(
+                    'The list: {} does not have files/folder. Attempting to fetch the file/folder content'.format(list))
+                response = self.sharepoint_client.get(rel_url, self.query)
+                if not response:
+                    logger.error(
+                        'Could not fetch the items for list: {}'.format(list_content))
+                    return []
+                try:
+                    response_data = response.json()
+                    response_data = response_data.get('d', {}).get('results')
+                except Exception as exception:
+                    logger.error('Error while parsing the get items response for list: {} from url: {}. Error: {}'.format(
+                        list_content, rel_url, exception))
+                    return []
+                logger.info(
+                    'Successfuly fetched and parsed the listitem response for list: {} from SharePoint'.format(list_content))
+
+            else:
+                logger.info(
+                    'The list: {} contains some files/folder'.format(list_content))
+                # TODO: Change the rel_url and query to fetch the file/folder contents
+                response_data = self.sharepoint_client.get(rel_url, self.query)
+                # TODO: need to add logic to index text
+                text = extract(response_data)
+
+            for index in range(len(response_data)):
+                response_data[index] = dict((k.lower(), v)
+                                            for k, v in response_data[index].items())
+                response_data[index].pop('__metadata')
+
+            if index:
```

Comment:
`index` is already defined before this `for` loop because it's a parameter of the current function.
at the end of the `for` loop the value of the `index` variable is overwritten and it will be the last value it received when executing the `for` loop.
This means that this `if` statement will always be executed.

A simplified example:
```
def index_items(index):
    for index in range(5):
        pass
    print("index is {}".format(index))
    if index:
        print("this if statement is always executed")

index_items(False)
```
outputs:
```
index is 4
this if statement is always executed
```

Reply:
Done